### PR TITLE
feat: Add depths to cell types for dot plot

### PR DESCRIPTION
**Pull request by pablo-gar**
- #2501

## Changes
- modify backend to create a tree structure for cell type ordering of scExpression
- update `generate_cell_ordering` from `transform.py` to write a pandas dataframe into a json file with the following columns `"tissue_ontology_term_id", "cell_type_ontology_term_id", "depth", "order"]`. The column `depth` encodes the level in the tree where a cell type is
- update `build_ordered_cell_types_by_tissue` of `v1.py` to work with the file mentioned above, the api request now returns a json file of the following structure `{tissue_id : [{id = "", name = "", depth = 10}, ... ], ...} `

### CODE REVIEW

File: backend/wmg/data/transform.py [hunk 1]
```diff
@@ -65,18 +65,25 @@ def compute_ordering(cells, root):
...
-        def recurse(node):
+        def recurse(node: Set[str], depth=0):
+
             if node in cells:
-                yield (node)
+
+                cells.remove(node)
+                yield {"id": node, "depth": depth}
+
+                if node != "CL:0000003":
```
Comment (MDunitz) [inline backend/wmg/data/transform.py:75]: why are we checking for this particular id?

File: backend/wmg/api/v1.py [hunk 2]
```diff
@@ -152,18 +153,49 @@ def build_ordered_cell_types_by_tissue(
...
+                "cell_type_ontology_term_id": row.cell_type_ontology_term_id,
+                "cell_type": ontology_term_label(row.cell_type_ontology_term_id),
+                "depth": row.depth,
+            }
         )
 
     return structured_result
+
+
+def build_ordered_cell_types_by_tissue_fix_depths(x):
```
Comment (MDunitz) [inline backend/wmg/api/v1.py:181]: can you use a more informative param name?

Comment (pablo-gar) [reply backend/wmg/api/v1.py]: done

**Review by MDunitz — APPROVED** [review]

File: backend/wmg/data/transform.py (hunk 1 above)
Comment (pablo-gar) [reply backend/wmg/data/transform.py:75]: "CL:0000003" is "naive cell", which we use for "unknown" cell types. However "naive cell" is the parent of virtually all other cell types, so we don't want put everything under naive cell.

In other words the CL structure looks like this:

-cell
---naive cell
------all other cells
------ [...]

but we want to make it look this way:
-cell
--naive celll
--all other cells
--[...]

**Pull request merged**